src/modules/ekf2/EKF/estimator_interface.cpp

```diff
@@ -65,29 +65,29 @@ void EstimatorInterface::setIMUData(const imuSample &imu_sample)
 		_initialised = init(imu_sample.time_us);
 	}
 
-	const float dt = math::constrain((imu_sample.time_us - _newest_high_rate_imu_sample.time_us) / 1e6f, 0.0001f, 0.02f);
+	_time_latest_us = imu_sample.time_us;
 
-	if (_newest_high_rate_imu_sample.time_us > 0) {
-		_dt_imu_avg = 0.8f * _dt_imu_avg + 0.2f * dt;
-	}
-
-	_newest_high_rate_imu_sample = imu_sample;
+	// the output observer always runs
+	_output_predictor.calculateOutputStates(imu_sample.time_us, imu_sample.delta_ang, imu_sample.delta_ang_dt, imu_sample.delta_vel, imu_sample.delta_vel_dt);
```

Comment:
We can still move it later, but as for the `init` function, it shouldn't be the job of `setIMUData` to call this function.